pkg/operations/cordondrainvm.go

```diff
@@ -34,14 +39,24 @@ func SafelyDrainNode(az armhelpers.ACSEngineClient, logger *log.Entry, masterURL
 	}
 
 	//Mark the node unschedulable
-	node, err := client.GetNode(nodeName)
-	if err != nil {
-		return err
-	}
-	node.Spec.Unschedulable = true
-	node, err = client.UpdateNode(node)
-	if err != nil {
-		return err
+	var node *v1.Node
+	for i := 0; i < cordonMaxRetries; i++ {
+		node, err = client.GetNode(nodeName)
+		if err != nil {
+			return err
+		}
+		node.Spec.Unschedulable = true
+		node, err = client.UpdateNode(node)
+		if err != nil {
+			// If this error is because of a concurrent modification get the update
+			// and then apply the change
+			if strings.Contains(err.Error(), kubernetesOptimisticLockErrorMsg) {
+				logger.Infof("Node %s got an error suggesting a concurrent modification. Will retry to cordon", nodeName)
+				continue
```

Comment:
Just curious whether we need to wait before retrying?

Reply:
I decided not to since this is a concurrent modification and I figured the other guy was done. We could wait a little in case they are making q bunch of small changes instead of a big one but I'm inclined to think it will usually be a batch update.